atlascharts/split-boxplot.js

```diff
@@ -0,0 +1,252 @@
+				// xAxisWidth bleeds over the allocated width.
+				tempXAxis.remove();				
+			}
+
+			let yAxisWidth = 0;
+	    if (options.showYAxis) {
+				// create temporary y axis
+				const tempYAxis = svg.append('g').attr('class', 'axis');
+				tempYAxis.call(yAxis);
+
+				// update height based on temp xaxis dimension and remove
+				yAxisWidth = Math.round(tempYAxis.node().getBBox().width);
+				width -= yAxisWidth;
+				tempYAxis.remove();
+			}
+	    // reset axis ranges
+	    x.range([0, width]);
+	    y.range([height, 0]);
+
+	    const boxHeight = options.boxHeight;
+	    let boxOffset = (y.bandwidth() / 2) - (boxHeight / 2);
+	    let whiskerHeight = boxHeight / 2;
+
+	    const chart = svg.append('g')
+	      .attr('transform', `translate(
+	          ${options.margins.left + yAxisWidth},
+	          ${options.margins.top}
+	        )`);
+
+	    // draw main box and whisker plots
+	    const boxplots = chart.selectAll('.boxplot')
+	      .data(data)
+	      .enter().append('g')
+	      .attr('class', 'boxplot')
+	      .attr('transform', d => `translate(0, ${y(d.Category)})`);
+
+	    const self = this;
+			
+			// set up scale for drawing box height
+			const boxScale = d3.scaleLinear()
+	      .range([boxHeight/2, 0])
+	      .domain([0,boxHeight]);
+			
+			const bandWidth = y.bandwidth();
+
+	    // for each g element (containing the boxplot render surface), draw the whiskers, bars and rects
+	    boxplots.each(function (boxplotData) {
+	      const boxplot = d3.select(this);
+				
+				const boxplotContainer = boxplot.append('g')
+					.attr('transform', () => `translate(0, ${boxOffset - 2})`);
+				
+				const targetBox = boxplotContainer.append('g')
+					.datum( boxplotData.target)
+					.attr('class', 'target');
+				const compareBox = boxplotContainer.append('g')
+					.datum(boxplotData.compare)
+					.attr('class', 'compare')
+					.attr('transform', () => `translate(0, ${boxHeight + 2}) scale(1,-1)`);
+				
+				let parts = [ 
+					{ "boxPlotData": boxplotData.target, "boxplot": targetBox},
+					{ "boxPlotData": boxplotData.compare, "boxplot": compareBox},
+				];
+				
+				parts.forEach(part => {
+					let d = part.boxPlotData;
+					let boxplot = part.boxplot;
+					
+					if (d.LIF != d.q1) { // draw whisker
+						boxplot.append('line')
+							.attr('class', 'bar')
+							.attr('x1', x(d.LIF))
+							.attr('y1', boxScale(0))
+							.attr('x2', x(d.LIF))
+							.attr('y2', boxScale(whiskerHeight));
+						boxplot.append('line')
+							.attr('class', 'whisker')
+							.attr('x1', x(d.LIF))
+							.attr('y1', boxScale(0))
+							.attr('x2', x(d.q1))
+							.attr('y2', boxScale(0));
+					}
+
+					boxplot.append('rect')
+						.attr('class', 'box')
+						.attr('x', x(d.q1))
+						.attr('y', boxScale(boxHeight))
+						.attr('width', Math.max(1, x(d.q3) - x(d.q1)))
+						.attr('height', boxScale(0) - boxScale(boxHeight))
+						.on('mouseover', d => self.tip.show(d, event.target))
```

Comment:
Tooltip should be aligned by top border of each box otherwise blinking appears
![image](https://user-images.githubusercontent.com/10010071/45114012-e2131400-b119-11e8-94c6-0c8f574d84e7.png)

Reply:
This is an artifact of having to apply a scale(1,-1) to the ```<g>``` element to use the same drawing instructiosn for the top boxplot as the bottom boxplot. What this does, however, is make ti look like a mirrior. I'm hoping we can live with this anomoly because the atlerative is writing the two boxplots with slightly different coordinates and you will fidn that the code looks nearly identical and will want me to try to refactor it (which I tried to reduce code-duplication in this case already)